client.go

```diff
@@ -101,9 +107,18 @@ func (c *Client) Dial(address string) (conn *Conn, err error) {
 	if useTLS {
 		network = strings.TrimSuffix(network, "-tls")
 
+		/*
+			// Enable after Go 1.18 is released, to be able to support two prev. releases.
```

Comment:
It would be a little clearer to move the single line comment here before the multi-line comment and prefix with the TODO.